src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -305,17 +306,38 @@ public boolean postString(final Locale locale, final String component,
         return cs.getMessages().getCachedData();
     }
     
-    /**
-     * Retrieves localized messages
-     *
-     * @param locale The locale in which the message is requested to be localized
-     * @param component The Singleton component 
-     * @return Message keys, each one mapped to the localized message
+     /**
+     * Retrieves the set of localized messages from the cache. It applies locale fallback mechanism in case of failure.
+     * 
+     * @param component The Singleton component
+     * @param locale The locale in which the messages are requested to be localized
+     * @return One of the items in the following priority-ordered list: 
+     * <ul>
+     * 		<li>The messages in the requested locale</li> 
+     * 		<li>The messages in the default locale</li>
+     * 		<li>The source messages</li>
+     * </ul>
      */
     public Map<String, String> getMessages(final Locale locale, final String component) {
         MessagesDTO dto = new MessagesDTO(component, null, null, locale.toLanguageTag(), this.cfg);
-        ComponentService cs = new ComponentService(dto);
-        return cs.getMessages().getCachedData();
+        MessageCacheItem cacheItem = new ComponentService(dto).getMessages();
+        
+        // While failed to get MessageCacheItem, use MessageCacheItem of the next fallback locale. 
+    	Iterator<Locale> fallbackLocalesIter = LocaleUtility.getFallbackLocales().iterator();
+    	while (cacheItem.getCachedData().isEmpty() && fallbackLocalesIter.hasNext()) {
+    		Locale fallback = fallbackLocalesIter.next();
+			MessagesDTO fallbackLocaleDTO = new MessagesDTO(dto.getComponent(), 
+					dto.getKey(), dto.getSource(), fallback.toLanguageTag(), null);
+			cacheItem = new ComponentService(fallbackLocaleDTO).getMessages();
+			
+			// Cache a reference to the MessageCacheItem of the fallback locale 
+			if (!cacheItem.getCachedData().isEmpty()) {
+				CacheService cacheService = new CacheService(dto);
+				cacheService.addCacheOfComponent(cacheItem);
```

Comment:
The loop should break here.

Reply:
If !cacheItem.getCachedData().isEmpty(), then the while loop condition is not satisfied anymore in the next loop. So it naturally breaks, no need to add a break statement.

Reply:
Don't need to judge twice in one cycle.

Reply:
I don't get it. What do you mean by judge twice in one cycle?